mirrord/layer/src/file/ops.rs

```diff
@@ -217,6 +219,80 @@ pub(crate) fn fdopen(fd: RawFd, rawish_mode: Option<&CStr>) -> Detour<*mut FILE>
     Detour::Success(result)
 }
 
+/// creates a directory stream for the `remote_fd` in the agent
+#[tracing::instrument(level = "trace")]
+pub(crate) fn fdopendir(fd: RawFd) -> Detour<usize> {
+    // usize == ptr size
+    // we don't return a pointer to an address that contains DIR
+
+    let open_files = OPEN_FILES.lock()?;
+    let remote_fd = open_files.get(&fd).ok_or(Bypass::LocalFdNotFound(fd))?;
+
+    let (dir_channel_tx, dir_channel_rx) = oneshot::channel();
+
+    let open_dir_request = OpenDir {
+        remote_fd: *remote_fd,
```

Comment:
Don't use same fd.. let the remote generate fd like we do for everything else.

Reply:
i am not using the same fd here, i am just letting the agent know which path it should refer to

Reply:
Didn't put it in correct location but point stands